module-4/backend/week_two.md

```diff
@@ -4,8 +4,56 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 
 Note: When you're done, submit a PR. 
 
-TBD  
+1. What's one difference between ES5 and ES6? 
+
+The way one declares a class/constructor function. I'm converted to Es6. I love js classes <3
+
+2. What's the difference between asynchronous and synchronous JavaScript? 
+
+Synchronous javascript executes one command at a time, in order. Asycnchronous javascript starts to execute a command, and then starts the next command. When the asynchronous command is completed, it reenters the queue to be completed.
+
+3. What are the four pillars of Object Oriented programming?
+
+Abstraction Polymorphic Inheritance Encapsulation (Thanks for the lighting talk, Ben Jacobs!)
+
+4. What are some tools available in JavaScript to help you write object oriented code?
+
+Classes, Static functions, Prototype functions, call back functions
+
+5. What are some key concepts to be aware of when refactoring your JavaScript?
+
+Single responsibility. Object oriented. Readability. Also, all the pillars.
+
+6. What's a callback function and what are some reasons when we use/need callback functions?
+
+A function passed into another function as an argument. We'd use them when we want to explicitly determine the order the functions are exected (i.e. the call back function would not be executed until the parent function was executed). Also, to keep code dry.
+
+7. What's the scope of variables in Javascript?
+
+local variables are available within functions, global variables are available outside of individual functions
```

Comment:
Gloabl vars are also available within functions within that global scope. SImilarly, if `functionb` is nested in `functionA`, `functionB` has access to `functionA` variables:

```js
function a () {
  var age = 10;
  function b () {
    console.log(age) // => 10 ( this works )
  }
}
```